src/NuGet.Core/NuGet.Common/ConcurrencyUtilities.cs

```diff
@@ -24,7 +24,7 @@ public static class ConcurrencyUtilities
             }
 
             // limit the number of unauthorized, this should be around 30 seconds.
-            var unauthorizedAttemptsLeft = 3000;
```

Comment:
There's a test I think to verify the correctness of the async locking mechanism. The same test should be executed on the sync version.